qualtran/bloqs/arithmetic/addition.py

```diff
@@ -69,7 +82,11 @@ def apply(self, *register_values: int) -> Union[int, Iterable[int]]:
     def on_classical_vals(
         self, a: 'ClassicalValT', b: 'ClassicalValT'
     ) -> Dict[str, 'ClassicalValT']:
-        return {'a': a, 'b': a + b}
+        assert self.bitsize <= 64  # TODO: be smarter
+        return {
+            'a': a,
+            'b': np.uint64(a) + np.uint64(b),
+        }  # TODO: account for signed integer addition
```

Comment:
for todo's please open an issue describing the problem and what a solution would look like and link to it in the todo comment